common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleBlockResolver.java

```diff
@@ -315,7 +301,7 @@ private ManagedBuffer getSortBasedShuffleBlockData(
           "shuffle_" + shuffleId + "_" + mapId + "_0.data"),
         shuffleIndexRecord.getOffset(),
         shuffleIndexRecord.getLength());
-    } catch (ExecutionException e) {
+    } catch (CompletionException e) {
```

Comment:
Can you tell me why we needed to change the exception? Is this just what Caffeine throws instead? Do we have test coverage for this?

Reply:
> Can you tell me why we needed to change the exception? Is this just what Caffeine throws instead? 

`com.github.benmanes.caffeine.cache.LoadingCache#get` method throw `CompletionException` if a checked exception was thrown while loading the value.

`com.google.common.cache.LoadingCache#get` method throw `ExecutionException` if a checked exception was thrown while loading the value.

So the exception type here has changed.

Reply:
> Do we have test coverage for this?

Because the `ExecutionException (now is CompletionException)` is re-throw as `RuntimeException`, I need to further check whether it is covered by existing test case

Reply:
@holdenk `testFetchWrongExecutor` and `testFetchNonexistent` in `ExternalBlockHandlerSuite` already cover this.